semantic_segmentation/code/helper.py

```diff
@@ -1,17 +1,18 @@
-import torch
-from torch.utils import data
-from torchvision import transforms
-import torchvision
-import zipfile
-from torchvision.datasets.utils import *
-import torch.nn.functional as F
 import numpy as np
-import os
-import cv2
-from collections import namedtuple
-from tqdm import tqdm
-from PIL import Image
-   
+
+
+def get_savename(hyper_params, mode=None, p=None):
+    if p is not None:
+        less = '/less_data/' + '_' + str(p) + '_'
+    else:
+        less = ''
+    if mode == 'stagewise':
+        return '../saved_models/' + less + hyper_params['model'] + '/stage' + str(hyper_params['stage'] - 1) + '/model' + str(
+            hyper_params['seed']) + '.pt'
+    if mode == 'classifier':
+        return '../saved_models/' + less + hyper_params['model'] + '/classifier/model' + str(hyper_params['seed']) + '.pt'
+
```

Comment:
might be good to have a worst case `return` statement i.e. if both if statements evaluate False (in case wrong arguments are passed).